### ISSUE TITLE
Add unit property to axis

### ISSUE THREAD

**Issue by mosc9575** [bokeh/bokeh#13469]
It would be nice to have a unit per axis to avoid redunant and user unfriendly ticking.

The figure below is an example where the unit is set per tick.

![grafik](https://github.com/bokeh/bokeh/assets/68053396/483829f8-427d-42f7-95e3-b92679c521f3)

What I have in mind is something like below:
![Unit-Tick](https://github.com/bokeh/bokeh/assets/68053396/90c77398-7c07-43e9-8781-e17ab47e2eef)

And this is how I would expect the potential code:

```
plot = figure(height=200)
plot.title = "Current over Resistance at a static voltage of 1 volt"
plot.circle(x, y, fill_color="blue", size=5)
plot.line(x, y, color="darkgrey")

plot.xaxis.axis_label = "Resistance"
plot.xaxis.unit = "$$\Omega$$"
plot.yaxis.axis_label = "Current"
plot.yaxis.unit = "$$A$$"
show(plot)
```

## Why is this useful?

I can think of some reasons:

1. Less use of `major_label_overrides` which simplifies the usage
2. Shorter ticks but keeping the same information
3. potential option to add timezone information to a datetime axis

**Comment by ianthomas23** [comment]
Personally I would put the units at the end of the axis label rather than show them in a separate location, i.e.

```python
plot.xaxis.axis_label = r"Resistance $$\Omega$$"
```

Some of the LaTeX examples are a bit contrived to show off LaTeX in tick labels, and I would never use them that way in real life. I would be more inclined to change those examples to put the units at the end of the axis label, now that we support inline math text (thanks for that, by the way!).

**Comment by mosc9575** [comment]
> Personally I would put the units at the end of the axis label rather than show them in a separate location, i.e.
> 
> ```python
> plot.xaxis.axis_label = r"Resistance $$\Omega$$"
> ```

I agree that is it straight forward to use the `axis_lable` and it is done in many examples, e.g. in the [transform_markers][https://docs.bokeh.org/en/latest/docs/examples/basic/data/transform_markers.html).

Nevertheless I like the idea move the unit to the same column/row where the actual ticks are.

**Comment by bryevdv** [comment]
"unit" seems like too narrow of a concept, and would artificially limit the utility of the feature. I'd be more inclined for this if the framing was for a generic "axis annotation" or similar (which could be a unit, but does not have to be).
